### PR TITLE
Hsm/signapi

### PR DESCRIPTION
**Pull request by musitdev**
# Summary
- **RFCs**: [Link to RFC](#./link/to/rfc), [Link to RFC](#./link/to/rfc), or $\emptyset$.
- **Categories**: any of `protocol-units`, `networks`, `scripts`, `util`, `cicd`, or `misc`.

First draft of the signing API as a lib.
# Changelog

# Testing

# Outstanding issues

### CODE REVIEW

File: util/signing/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#[derive(Debug, Clone, Default)]
+pub struct KeyVersion(pub String);
+
+/// Id that identify a Key.
+#[derive(Debug, Clone)]
+pub struct KeyId(pub String);
+
+pub struct SigningService;
+
+impl SigningService {
```
Comment (mzabaluev) [inline util/signing/src/lib.rs:24]: Should this rather be a trait? I think we need implementations of this for the different backends.

File: util/signing/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	/// Create the service with environment variable.
+	pub fn try_from_env() -> Result<Self, anyhow::Error> {
```
Comment (mzabaluev) [inline util/signing/src/lib.rs:26]: If `SigningService` (or `Hsm` in the earlier demo PRs) is to be a trait, this does not belong in the abstraction. We potentially want the dynamic dispatch to be feasible, and the constructor APIs should belong with the specific backends.

**Comment by l-monninger** [comment]
Right now, this looks more less like what was already used in #961 and proposed in the original demo application. 

I recommend you do a spec API similar to #961 to make it easier to generate modules. Check the following lines for reference:
- `AwsKms` spec: https://github.com/movementlabsxyz/movement/blob/l-monninger/hsm-demo/demo/hsm/src/cryptography/aws_kms.rs
- `AwsKms` spec usage: https://github.com/movementlabsxyz/movement/blob/df28163913abe8d88d9510fd2099eeeba50bdc0d/demo/hsm/src/hsm/aws_kms.rs#L81
- `AwsKms` HSM usage with spec: https://github.com/movementlabsxyz/movement/blob/2093ac1e3530fffbcffa8b3386067929155c5146/demo/hsm/src/cli/server/secp256k1/aws_kms.rs#L14

File: util/signing/src/lib.rs (hunk 1 above)
Comment (musitdev) [reply util/signing/src/lib.rs:39]: The idea was to be backend agnostic so that the application don't have to know how keys are organized.

**Pull request closed without merging**